packages/documentation/src/stories/components/togglebutton/togglebutton.stories.ts

```diff
@@ -0,0 +1,160 @@
+
+export interface PostTogglebuttonProps {
+  type?: 'button' | 'submit' | 'reset';
+  toggled?: boolean;
+  variant?: string;
+  size?: string;
+  contentWhenToggled?: string;
+  contentWhenUntoggled?: string;
+}
+
+const meta: MetaComponent<PostTogglebuttonProps> = {
+  id: '1a6f47c2-5e8a-45a0-b1c3-9f7e2b834c24',
+  title: 'Components/Toggle Button',
+  tags: ['package:WebComponents'],
+  component: 'post-togglebutton',
+  parameters: {
+    design: {},
+  },
+  args: {
+    contentWhenToggled: 'Toggled',
+    contentWhenUntoggled: 'Untoggled',
+    type: 'button',
+    variant: 'btn-primary',
+    size: 'null',
+    toggled: false,
+  },
+  argTypes: {
+    toggled: {
+      control: 'boolean',
+      description: 'Defines whether the button is toggled or not.',
+    },
+    contentWhenToggled: {
+      name: 'Toggled text',
+      description: "Text within the button when it's toggled.",
+      control: 'text',
+      table: {
+        category: 'Content',
+      },
+    },
+    contentWhenUntoggled: {
+      name: 'Untoggled text',
+      description: "Text within the button when it's untoggled.",
+      control: 'text',
+      table: {
+        category: 'Content',
+      },
+    },
+    type: {
+      name: 'Input type',
+      description: 'Defines the input type.',
+      if: {
+        arg: 'tag',
+        eq: 'input',
+      },
+      control: {
+        type: 'inline-radio',
+        labels: {
+          button: 'Button',
+          submit: 'Submit',
+          reset: 'Reset',
+        },
+      },
+      options: ['button', 'submit', 'reset'],
+      table: {
+        category: 'General',
+      },
+    },
+    size: {
+      name: 'Size',
+      description:
+        'Sets the size of the component.' +
+        '<span className="mt-8 alert alert-info alert-sm">' +
+        'For more options, please see the ' +
+        '<a href="/?path=/docs/eb78afcb-ce92-4990-94b6-6536d5ec6af4--docs">button component</a>' +
+        '.</span>',
+      control: {
+        type: 'select',
+        labels: {
+          'btn-sm': 'Small',
+          'btn-rg': 'Regular',
+          'null': 'Medium',
+          'btn-lg': 'Large',
+        },
+      },
+      options: ['btn-sm', 'btn-rg', 'null', 'btn-lg'],
+      table: {
+        category: 'General',
+      },
+    },
+    variant: {
+      name: 'Variant',
+      description:
+        'Defines a style variant.' +
+        '<span className="mt-8 alert alert-info alert-sm">' +
+        'For more options, please see the ' +
+        '<a href="/?path=/docs/eb78afcb-ce92-4990-94b6-6536d5ec6af4--docs">button component</a>' +
+        '.</span>',
+      control: {
+        type: 'inline-radio',
+        labels: {
+          'btn-primary': 'Primary',
+          'btn-secondary': 'Secondary',
+          'btn-tertiary': 'Tertiary',
+        },
+      },
+      options: ['btn-primary', 'btn-secondary', 'btn-tertiary'],
+      table: {
+        category: 'General',
+      },
+    },
+  },
```

Comment:
`...buttonMeta.argTypes.variant` I can't do this, i have this error:

`TS18048: meta.argTypes is possibly undefined`

Reply:
This is because `buttonMeta` is of type `Meta` and the `argTypes` is optional on this type which means it can sometimes be undefined. Can you try with `...buttonMeta.argTypes?.variant`? The `?.` mean "if `argTypes` is defined then returns its `variant` property, otherwhise return undefined".